### DISCUSSION TITLE
Cargo Scooters and selecting weapons

### DISCUSSION THREAD

**Discussion by Grenno9** [FakeFishGames/Barotrauma#11698] [Ideas and suggestions]
This is probably more a feature request than a bug, but currently the way that cargo scooters are handled are really cumbersome.

When you're in combat and holding a cargo scooter, if you go to select a gun, the gun is placed into the cargo scooter rather than activated. 

You have to drag off the cargo scooter with your mouse and THEN activate the gun. This is really counterintuitive, especially when you're getting accosted by a bunch of spinelings. Especially as 9/10 times you have already clicked and now you need to go into the cargo scooter, take then gun out and then drop the cargo scooter and activate the gun.

Would it be possible to automatically drop the cargo scooter if you go to select a weapon? Or even better, have a key binding that drops whatever you are currently holding 😃

Cheers!

**Comment by Prolbo** [comment]
Or dedicated "drop whatever the heck you currently holding " button.

**Comment by IcarusMJ12** [comment]
If it helps I just made a [mod](https://steamcommunity.com/sharedfiles/filedetails/?id=2984665813) for this.

**Comment by 8on8er** [comment]
Hard agree, I think putting weapons into a cargo scooter should be done by double clicking not selecting the weapon.

**Comment by pavel4win** [comment]
Here is mod to drop anything you hold. Mod is client side, need lua.
https://steamcommunity.com/sharedfiles/filedetails/?id=2768813474
Very useful

**Comment by Next-Door-Tech** [comment]
Gently bumping this. Was going to create a new suggestion to the exact same effect - selecting tools or weapons while a cargo scooter is equipped should drop the cargo scooter and equip the tool/weapon instead of storing the item.

IMO, the "drop what you're holding button" solution is a nice extra addition, but I'd still prefer the cargo scooter behavior be changed as well.

**Reply by IcarusMJ12** [reply]
> Gently bumping this. Was going to create a new suggestion to the exact same effect - selecting tools or weapons while a cargo scooter is equipped should drop the cargo scooter and equip the tool/weapon instead of storing the item.
> 
> IMO, the "drop what you're holding button" solution is a nice extra addition, but I'd still prefer the cargo scooter behavior be changed as well.

Consider the mod I posted above -- it still works AFAICT.
